### PR TITLE
[BTS-721] Added distinguishing warning messages for query options index hint

### PR DESCRIPTION
**Pull request by cpjulia**
### Scope & Purpose

When creating a query that contains non accepted OPTIONS, the warning message raised to showcase was too broad. Now, the warning message raised will distinguish for the user between when the OPTIONS attribute is correct, but the value is incorrect, and when the attribute is incorrect. Example:
```
arangod> db._query(`FOR doc IN myCollection OPTIONS {forceIndexHints: "abcd"} FILTER doc.value1 != "null" RETURN doc`)
[object GeneralArrayCursor, count: 0, cached: false, warning(s): "1539 usage of unknown OPTIONS attribute 'forceIndexHints' in FOR statement"]
arangod> db._query(`FOR doc IN myCollection OPTIONS {forceIndexHint: 1} FILTER doc.value1 != "null" RETURN doc`)
[object GeneralArrayCursor, count: 0, cached: false, warning(s): "1539 usage of invalid value 1 in  OPTIONS attribute 'forceIndexHint' in FOR statement"]
```

- [x] :hankey: Bugfix (not exactly a bugfix, but a fix to the warning)
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [x] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [x] Forward ports
   Forward port devel https://github.com/arangodb/arangodb/pull/16781
   Forward port 3.10 https://github.com/arangodb/arangodb/pull/16780

#### Related Information

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [x] GitHub issue / Jira ticket: https://arangodb.atlassian.net/browse/BTS-720
- [ ] Design document:

### CODE REVIEW

File: arangod/Aql/IndexHint.cpp [hunk 1]
```diff
@@ -141,10 +141,16 @@ IndexHint::IndexHint(QueryContext& query, AstNode const* node) : IndexHint() {
...
+          ExecutionPlan::invalidOptionAttribute(query, "unknown", "FOR",
+                                                name.data(), name.size());
+          handled = true;
         }
-
         if (!handled) {
-          ExecutionPlan::invalidOptionAttribute(query, "unknown", "FOR",
+          VPackBuilder builder;
+          child->getMember(0)->toVelocyPackValue(builder);
+          std::string msg = "invalid value " + builder.toString() + " in ";
```
Comment (jsteemann) [inline arangod/Aql/IndexHint.cpp:152]: I'd prefer using `toJson()` here, as `toString()` will contain pretty pretting, which is not only verbose, but also less readable when contained inside another error message/warning.
```suggestion
          std::string msg = "invalid value " + builder.toJson() + " in ";
```

**Review by jsteemann — APPROVED** [review]
LGTM

**Comment by KVS85** [comment]
Only unrelated known failure on Windows in Jenkins.

**Review by KVS85 — APPROVED** [review]
LGTM.

**Pull request merged**